src/components/views/SignUp.jsx

```diff
@@ -33,6 +34,14 @@ class SignUp extends Component {
   }
 
   submit({ password }) {
+    // TODO: move this some place better
+    const createAppropriateWallet = (provider, pwd) => {
+      if (typeof web3 !== 'undefined') {
+        return BaseWallet.createWallet(provider, pwd);
+      }
+      return GivethWallet.createWallet(provider, pwd);
+    };
+
```

Comment:
so the idea is to have a `Wallet` interface & then implementations of that interface so we don't need to do things like this.

Instead we would just pass the `Wallet` instance to the components and call the methods directly